scripts/submodule-diff.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
```

Comment:
Consider adding `set -euxo pipefail` (https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail)

Reply:
I think for this use-case it isn't what we want, because it would exit the script and Github Action early and we would not get output for any submodules that were not checked yet.

Reply:
I would still put it. We want to exit whenever a command fails. Consider 4 fails (silently). Then we would get an empty diff.

Reply:
add a small summary of what the script does.

Reply:
`diff` returns nonzero upon any difference. So no, here we don't want to exit upon any 'fail', we'd ever only get the difference reported for a single module.

I'll add a summary!

Reply:
You're right (Cf. http://mywiki.wooledge.org/BashFAQ/105). But what about error handling, e.g., if `git submodule foreach 'git fetch'` fails?

Reply:
When I disable network and run the script, the `fetch` loop terminates early (after the first remote being inaccessible) and the diff is generated based on unupdated module remotes. That is not terribly useful, but also not terribly bad, assuming that by running this weekly, such a failure every now and then will do no more harm than postpone us being notified a week late of updates.

Is that acceptable?